### ISSUE TITLE
Defining System.Void in compilation suppresses error about usage of System.Void

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#61500]
**Version Used**: e6b5dd830f1b790bd80e62272129cc040d0a2fdc

**Steps to Reproduce**:

[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTMJgKYDOADhAMZ4AEC6GA3phY1QMxVIUBqA9lACYW0KAXwoB6URVJcw+GABcKXAK4K5XCgHM8CiDAp4ATga4GKsAnLwR+XAGYUIFAO4QDMWBsxDMmBK2QUAMICDEx+VAAsFACyABTcfBQAbgCUAozizq7uMBpBAMoADBHMAGwgFAAqABaUcgCeRJQA5Am8zWZ6zQD6HVIwtgA2UKRyBM5QctUUU5RQYEQmlvwNTRTNCBHNoYzeGEJAA=)

```cs
using System;

namespace System
{
    public class Void { } // comment out to get an error instead of a warning
}

public class C {
    public void M(Void v) {  // warning CS0436: The type 'Void' in '_' conflicts with the imported type 'void'
    }
}
```

**Expected Behavior**: An error is reported regardless of whether `Void` is defined in source.

**Actual Behavior**: An error is only reported when `Void` is not defined in source.

**Comment by jcouv** [comment]
Good find. 
However, this is likely an old issue and seems niche. Moving to backlog. Feel free to raise priority if appropriate.

**Comment by gafter** [comment]
I suspect this is not a bug.  Defining your own `System.Void` doesn't define the one that `void` refers to - that type is always the `System.Void` that comes from the "core library".  (Your assembly isn't the core library)
